src/3p.js

```diff
@@ -109,17 +109,20 @@ function executeAfterWriteScript(win, fn) {
  * @param {string} src
  */
 export function validateSrcPrefix(prefix, src) {
-
   if (!isArray(prefix)) {
     prefix = [prefix];
   }
-
-  for (const p of prefix) {
-    if (src.indexOf(p) === 0) {
-      return;
+  const totalPrefixes = prefix.length;
+  if (src !== undefined) {
+    let p = 0;
+    for (p = 0; p <= totalPrefixes; p++) {
+      const protocolIndex = src.indexOf(prefix[p]);
+      if (protocolIndex == 0) {
+        return;
+      }
     }
+    throw new Error('Invalid src ' + src);
```

Comment:
the first throw is if src is defined and no matches are found. The last throw is if src is undefined, try the sample below in the console which has the first throw removed, you'll see that the error is not throw when http does not exist in https://example.org

```
var toString_ = Object.prototype.toString;

function toString(value) {
  return toString_.call(value);
}

function isArray(value) {
  return toString(value) === '[object Array]';
}

function validateSrcPrefix(prefix, src) {
  if (!isArray(prefix)) {
    prefix = [prefix];
  }
  const totalPrefixes = prefix.length;
  if (src !== undefined) {
    for (var p = 0; p <= totalPrefixes; p++) {
      const protocolIndex = src.indexOf(prefix[p]);
      if (protocolIndex == 0) {
        return;
      }
    }
  }
  throw new Error('Invalid src ' + src);
}
```